src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -53,6 +55,45 @@ public void equals() {
         assertFalse(filterFirstCommand.equals(filterSecondCommand));
     }
 
+    @Test
+    public void execute_noTagFilteredFor_showsNothing() {
+        expectedModel.updateFilteredPersonList(person -> false);
+        String[] absentTags = {
+            "bestFriends",
+            "Colleagues",
+            "4Horsemen"
+        };
+        for (String absentTag : absentTags) {
+            FilterCommand filterCommand = new FilterCommand(new Tag(absentTag));
+            assertCommandSuccess(
+                    filterCommand,
+                    model,
+                    String.format(FilterCommand.MESSAGE_SUCCESS, absentTag),
+                    expectedModel
+            );
+        }
+    }
+
+    @Test
+    public void execute_tagFilteredForPresent_showsFiltered() {
+        String[] presentTags = {
+            "owesMoney",
+            "friends"
+        };
+        for (String presentTag : presentTags) {
+            FilterCommand filterCommand = new FilterCommand(new Tag(presentTag));
+            expectedModel.updateFilteredPersonList(
+                    person -> person.getTags().contains(new Tag(presentTag))
+            );
+            assertCommandSuccess(
+                    filterCommand,
+                    model,
+                    String.format(FilterCommand.MESSAGE_SUCCESS, presentTag),
+                    expectedModel
+            );
+        }
+    }
+
     /**
      * A default model stub that have all of the methods failing.
      */
```

Comment:
Perhaps ModelStub is not needed as it is unused here?

Reply:
that's a nice catch! i'll remove it before merging